### ISSUE TITLE
byra / redrix : missing Intel VT related options?

### ISSUE THREAD

**Issue by aoberoi** [MrChromebox/firmware#575]
first of all, thanks for this project. it's an incredible undertaking and it enables so much good. also, i'm a noob when it comes to firmware things, but i'd like to learn, so i appreciate your patience.

I have MrChromebox-4.22.2 installed on a Redrix. I don't see any options in the boot menu mentioning virtualization being enabled or disabled. Based on reading previous issues, its my understanding that the firmware installation scripts would enable virtualization options if the machine supported it. However, this machine has the Intel i7-1265U CPU, which as far as I can tell, [does support various Intel VT technologies](https://www.intel.com/content/www/us/en/products/sku/226258/intel-core-i71265u-processor-12m-cache-up-to-4-80-ghz/specifications.html). is my understanding incorrect? if not, is there anything i can do to help land a change to enable the virtualization option in the firmware?

for background: i have Windows 11 installed, and i was trying to set up WSL2. it seems to have failed because I get a BSOD with the stop code `INACCESSIBLE_BOOT_DEVICE` on reboot. booting into Safe Mode seems to revert the change and then I can boot up normally. when i tried to investigate, my only lead was part of the output from the `wsl --status` command:

```
WSL2 is not supported with your current machine configuration.
Please enable the "Virtual Machine Platform" optional component and ensure virtualization is enabled in the BIOS.
Enable "Virtual Machine Platform" by running: wsl.exe --install --no-distribution
For information please visit https://aka.ms/enablevirtualization
```

this led me to look into BIOS settings and realize that virtualization wasn't present. i investigated a bit more and it seems like virtualization is enabled (but I'm not sure what else I might be missing). Based on [this doc](https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/system-requirements-for-hyper-v-on-windows) on Hyper V requirements, the output of `Systeminfo`:

```
Hyper-V Requirements:      VM Monitor Mode Extensions: Yes
                           Virtualization Enabled In Firmware: Yes
                           Second Level Address Translation: Yes
                           Data Execution Prevention Available: Yes
```

so this is quite confusing, but it seems [i am not alone](https://github.com/microsoft/WSL/issues/4784). however most of the people who have succeeded in working around the issue cite enabling and disabling various BIOS/UEFI options including those related to virtualization, **and also a Kernel DMA Protection option**.

(Edit: add details from Systeminfo and interest in Kernel DMA Protection)

**Comment by MrChromebox** [comment]
> this led me to look into BIOS settings and realize that virtualization wasn't present.

there is no setting because it's not user configurable. VT-x and VT-d are both enabled on all parts which support it. You can verify this by looking at the CPU info on the Performance tab in Task Manager, or running CPU-z, or any other number of utilities. Your issue is not a firmware setting.

**Comment by aoberoi** [comment]
thanks for the quick response.

as you mentioned, i can see virtualization is enabled. in CPU-z, i see VT-x, but not VT-d. when you say these are not user configurable, does that mean the hardware is locked at VT-x enabled and VT-d disabled? or would it be possible to expose an option in the UEFI settings to toggle those settings? i've been googling around and all i've seen is people sharing screenshots of where its configured in their UEFI/BIOS, i hadn't considered that my hardware was not configurable.

**Comment by MrChromebox** [comment]
> when you say these are not user configurable, does that mean the hardware is locked at VT-x enabled and VT-d disabled?

it means that coreboot automatically enables those items if the hardware supports them, and there are no user options to enable/disable any features like you might find in a proprietary UEFI firmware. if your CPU supports

> all i've seen is people sharing screenshots of where its configured in their UEFI/BIOS, i hadn't considered that my hardware was not configurable.

you're not running anything close to the same firmware, what others can do is irrelevant. The hardware is configurable. The firmware is not (not without rebuilding).

All that said, I'll double check and make sure that the Alderlake SoC code is correctly setting and locking the VMX register bits

**Comment by coolstar** [comment]
WSL2 and VMWare run fine on brya / banshee here. You'll need to search in start menu for "turn windows features on or off" and enable "Virtual Machine Platform"

**Comment by aoberoi** [comment]
> it means that coreboot automatically enables those items if the hardware supports them, and there are no user options to enable/disable any features like you might find in a proprietary UEFI firmware. 

i understand that there are currently no user options. my question is whether or not user options can be added (in theory, at least)? if so, in which project(s) would that be done?

**Comment by MrChromebox** [comment]
> my question is whether or not user options can be added (in theory, at least)? if so, in which project(s) would that be done?

I'm not going to add pointless options. There's no need ever to disable VT-x or VT-d, and I have no desire for coreboot to look or act like some POS proprietary UEFI firmware. 

Adding any firmware config options requires doing so in both coreboot and edk2, and having some way to tie them together.

**Comment by aoberoi** [comment]
Okay, that's reasonable, thanks for considering.

My use case is simply to try out this workaround to what seems like a popular issue for Windows users (not limited to Chromebox firmware, but overlapping). Simply a vote in favor of some more configuration. Thanks!

**Comment by xiaojianqaq** [comment]
@aoberoi
I had the same issue in Windows 11 23H2, but it was resolved after I switched to the official 25H2 build. The system no longer crashes to a BSOD and boots normally. I hope this helps you even after all this time.

**Comment by aoberoi** [comment]
@xiaojianqaq that's really helpful input! i haven't been able to upgrade to 25H2 (still running 23H2) because I'm failing the secure boot system requirement check. Any advice for how to get past that?

UPDATE: I've now got WSL2 working on this machine. I was able to upgrade Windows using [this workaround](https://github.com/Win11Modder/Win11-Req-Bypass). I selected to tweak the registry, then selected to use a custom target version, then typed in 25H2. After the script was complete, I used Windows 11 Media Creation to create an .iso. I mounted that image, and proceeded with the Windows upgrade. After a long while and a restart, I was able to install WSL2.
